test/framework/taints.go

```diff
@@ -0,0 +1,54 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+func (e *ClusterE2ETest) ValidateNodeTaints(expectedTaints map[corev1.Taint]int) {
```

Comment:
Why put receiver method here instead of in cluster.go? Is there a general practice we follow for when we place receivers outside of the the file containing the type the receiver belongs too?

Reply:
in the case of the test framework we're using the file to differentiate and organize the framework functionality (e.g. taints specific test functionality in `taints.go`, OIDC in `oidc.go`) while still making it really straight forward to run the validations on the given test execution

Reply:
So `ValidateNodeTaints` does not validate if the taints count and value are correct on the worker node group level / control plane level?